### PR TITLE
Design for validating repository controller secrets

### PR DESCRIPTION
**Pull request by kale-amruta**
## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: design/kanister-kopia-integration.md [hunk 1]
```diff
@@ -599,3 +600,40 @@ credentials. This model ensures Kanister remains free from a hard dependency on
...
+- `secrets.kanister.io/kopia-repository/serveradmin`
+- `secrets.kanister.io/kopia-repository/serveruser`
+- `secrets.kanister.io/azure-location`
+- `secrets.kanister.io/gcp-location`
+- `secrets.kanister.io/aws-location`
+- `kubernetes.io/tls`
+- `secrets.kanister.io/aws`
+- `secrets.kanister.io/azure`
+- `secrets.kanister.io/gcp`
+
```
Comment (kale-amruta) [inline design/kanister-kopia-integration.md:637]: @pavannd1

File: design/kanister-kopia-integration.md [hunk 2]
```diff
@@ -599,3 +600,40 @@ credentials. This model ensures Kanister remains free from a hard dependency on
 any crypto packages, and vault-like functionalities.
 
 If misplaced, Kanister will not be able to recover these credentials.
+
+#### secrets-validation
+Approaches:
+
+1. Use `kanctl` to validate the secrets before they are created
+`kanctl` has a utility to validate resources. A new resource under `kanctl validate` command for secrets will be added. The command will also have a flag to validate what secret we need to validate.
```
Comment (ankitjain235) [inline design/kanister-kopia-integration.md:608]: If we see rich diff, both the lines are appearing together.

File: design/kanister-kopia-integration.md [hunk 3]
```diff
@@ -599,3 +600,40 @@ credentials. This model ensures Kanister remains free from a hard dependency on
...
+- `secrets.kanister.io/azure-location`
+- `secrets.kanister.io/gcp-location`
+- `secrets.kanister.io/aws-location`
+- `kubernetes.io/tls`
+- `secrets.kanister.io/aws`
+- `secrets.kanister.io/azure`
+- `secrets.kanister.io/gcp`
+
+2. Use admission controller to validate the secrets
+    We already have a validation webhook configured to validate the blueprint. We can add another endpoint which validates the secret. We will differentiate the repository server secrets with other secrets by using a kubernetes label
```
Comment (ankitjain235) [inline design/kanister-kopia-integration.md:639]: Same thing I can see here as well.

**Review by ankitjain235 — COMMENTED** [review]
Looks good overall, but just looking at the document, it's a bit unclear which approach was finalised. Should we mention that somewhere?

File: design/kanister-kopia-integration.md (hunk 3 above)
Comment (kale-amruta) [reply design/kanister-kopia-integration.md:639]: I will verify it

File: design/kanister-kopia-integration.md (hunk 2 above)
Comment (kale-amruta) [reply design/kanister-kopia-integration.md:608]: I will verify it

**Comment by kale-amruta** [comment]
Closing this PR

**Pull request closed without merging**